src/test/java/org/jenkinsci/plugins/docker/workflow/RegistryEndpointStepTest.java

```diff
@@ -181,6 +204,70 @@ public void stepExecutionWithCredentialsAndQueueItemAuthenticator() throws Excep
+                CredentialsProvider.lookupStores(r.jenkins).iterator().next().addCredentials(Domain.global(), registryCredentials);
+                var p = r.createProject(WorkflowJob.class, "p");
+                p.setDefinition(new CpsFlowDefinition(
+                    """
+                    node('remote') {
+                      mockDockerLogin {
+                        withDockerRegistry(url: 'https://my-reg:1234', credentialsId: 'registryCreds') {
+                          semaphore 'wait'
+                        }
+                      }
+                    }
+                    """, true));
+                var b = p.scheduleBuild2(0).waitForStart();
+                SemaphoreStep.waitForStart("wait/1", b);
+            });
+            @SuppressWarnings("deprecation")
+            var verboten = FilePathPickle.class.getName();
+            assertThat(StringEscapeUtils.escapeJava(Files.readString(rr.getHome().toPath().resolve("jobs/p/builds/1/program.dat"), StandardCharsets.ISO_8859_1)), not(containsString(verboten)));
+            rr.then(r -> {
+                var b = r.jenkins.getItemByFullName("p", WorkflowJob.class).getBuildByNumber(1);
+                SemaphoreStep.success("wait/1", null);
+                r.assertBuildStatusSuccess(r.waitForCompletion(b));
+                r.assertLogContains("docker login -u me -p ******** https://my-reg:1234", b);
+                r.assertLogNotContains("s3cr3t", b);
+            });
+        }
+    }
+
+    private static final class SSHAgentContainer extends GenericContainer<SSHAgentContainer> {
+        private final String priv;
+        SSHAgentContainer() {
+            super("jenkins/ssh-agent:6.17.0");
+            try {
+                var kp = KeyUtils.generateKeyPair(KeyPairProvider.SSH_RSA, 2048);
+                var kprw = new OpenSSHKeyPairResourceWriter();
+                var baos = new ByteArrayOutputStream();
+                kprw.writePublicKey(kp, null, baos);
+                var pub = baos.toString(StandardCharsets.US_ASCII);
+                baos.reset();
+                kprw.writePrivateKey(kp, null, null, baos);
+                priv = baos.toString(StandardCharsets.US_ASCII);
+                withEnv("JENKINS_AGENT_SSH_PUBKEY", pub);
+                withExposedPorts(22);
+            } catch (Exception x) {
+                throw new AssertionError(x);
+            }
+        }
+        void register(String name) throws Exception{
+            var creds = new BasicSSHUserPrivateKey(CredentialsScope.GLOBAL, null, "jenkins", new BasicSSHUserPrivateKey.DirectEntryPrivateKeySource(priv), null, null);
+            CredentialsProvider.lookupStores(Jenkins.get()).iterator().next().addCredentials(Domain.global(), creds);
+            var port = getMappedPort(22);
+            Jenkins.get().addNode(new DumbSlave(name, "/home/jenkins/agent", new SSHLauncher(getHost(), port, creds.getId())));
```

Comment:
(For use with `RealJenkinsRule`, would need a bit of refactoring to bind `name`, `host`, `port`, and `priv` in a `record`.)